### PR TITLE
5200 upgradable contractGovernor

### PR DESCRIPTION
**Pull request by turadg**
refs: #5200 

## Description

This makes the `contractGovernor` contract upgradable. It's not yet fully durable because some facets aren't durable and the open questions can't be validated after a contract restart/upgrade, so I'm leaving #5200 open.

It does satisfy the requirement that each governor contract vat be upgradable,
- https://github.com/Agoric/agoric-sdk/issues/6553

### Security Considerations

Authority from baggage

### Scaling Considerations

Should lower memory pressure slightly

### Documentation Considerations

Little change to API. The `voteOnApiInvocation` is now async, so that it can lazily query the remote governed contract for its API methods.

### Testing Considerations

Existing tests pass. Bootstrap test of vaults upgrade now upgrades the vaultFactory's governor.

### CODE REVIEW

File: packages/governance/src/contractGovernance/governApi.js
Comment (dckc) [inline packages/governance/src/contractGovernance/governApi.js:41]: these weren't used? how did that pass lint?

File: packages/governance/src/contractGovernance/governFilter.js
Comment (dckc) [inline packages/governance/src/contractGovernance/governFilter.js:99]: is this a breaking contract API change? Is it required?

File: packages/governance/src/contractGovernor.js
Comment (dckc) [inline packages/governance/src/contractGovernor.js]: accidental space?

File: packages/governance/src/contractGovernor.js
Comment (dckc) [inline packages/governance/src/contractGovernor.js]: add `governedContractLabel` to this trace?

File: packages/governance/src/contractGovernor.js
Comment (dckc) [inline packages/governance/src/contractGovernor.js]: So this is not commenting out dead code but telling future maintainers "you may think to do it this way; don't." right?

File: packages/governance/src/contractGovernorKit.js
Comment (dckc) [inline packages/governance/src/contractGovernorKit.js:1]: I think the tests are better at judging whether the logic was preserved when moving this to a new file in exo form than I would be, so I'm just skimming.

File: packages/vats/test/bootstrapTests/test-vaults-upgrade.js
Comment (dckc) [inline packages/vats/test/bootstrapTests/test-vaults-upgrade.js:220]: :clap:

**Review by dckc — COMMENTED** [review]
the Far change looks like a breaking contract api change. I'd like to know if I'm on target about that before approving.

It's probably worth waiting on a review by @Chris-Hibbert too

File: packages/governance/src/contractGovernance/governApi.js
Comment (turadg) [reply packages/governance/src/contractGovernance/governApi.js:41]: because the subsequent params were used, they had to stay there for backwards compatibility

File: packages/governance/src/contractGovernance/governFilter.js
Comment (turadg) [reply packages/governance/src/contractGovernance/governFilter.js:99]: an internal refactor: [refactor(governor): consistent createdQuestion](https://github.com/Agoric/agoric-sdk/pull/7657/commits/c692fcc5ce2f6683357aa27e31bac04873d3434d)

File: packages/governance/src/contractGovernor.js
Comment (turadg) [reply packages/governance/src/contractGovernor.js]: yeesh yes

File: packages/governance/src/contractGovernor.js
Comment (turadg) [reply packages/governance/src/contractGovernor.js]: correct. I can clarify

File: packages/governance/src/contractGovernance/governFilter.js
Comment (dckc) [reply packages/governance/src/contractGovernance/governFilter.js:99]: why is the object it's in `Far`?

Comment (turadg) [reply packages/governance/src/contractGovernance/governFilter.js:99]: good question. I don't really know. I do know that what is returned does not exposed outside the `contractGovernorKit`. But given the time pressure I'm content to strike the change. 

If you approve contingent on that I'll omit it in the rebase I have to do.

Comment (dckc) [reply packages/governance/src/contractGovernance/governFilter.js:99]: I looked around and i don't see it escaping, so I agree it's internal.

**Review by dckc — APPROVED** [review]
I'd like to hear from @Chris-Hibbert on this, but I'm ok if we don't.

File: packages/governance/src/contractGovernor.js
Comment (Chris-Hibbert) [inline packages/governance/src/contractGovernor.js]: revert dropping false.

File: packages/governance/src/contractGovernorKit.js
Comment (Chris-Hibbert) [reply packages/governance/src/contractGovernorKit.js:1]: I did more than skim. I don't see any errors or omissions. I think it's all here.

**Review by Chris-Hibbert — APPROVED** [review]
lgtm

**Pull request merged**
